src/material/snack-bar/testing/shared.spec.ts

```diff
@@ -0,0 +1,166 @@
+    expect(await snackBar.getRole()).toBe('status');
+
+    fixture.componentInstance.openCustom({politeness: 'off'});
+    snackBar = await loader.getHarness(snackBarHarness);
+    expect(await snackBar.getRole()).toBe(null);
+  });
+
+  it('should be able to get message of simple snack-bar', async () => {
+    fixture.componentInstance.openSimple('Subscribed to newsletter.');
+    let snackBar = await loader.getHarness(snackBarHarness);
+    expect(await snackBar.getMessage()).toBe('Subscribed to newsletter.');
+
+    // For snack-bar's with custom template, the message cannot be
+    // retrieved. We expect an error to be thrown.
+    fixture.componentInstance.openCustom();
+    snackBar = await loader.getHarness(snackBarHarness);
+    await expectAsyncError(() => snackBar.getMessage(), /custom content/);
+  });
+
+  it('should be able to get action description of simple snack-bar', async () => {
+    fixture.componentInstance.openSimple('Hello', 'Unsubscribe');
+    let snackBar = await loader.getHarness(snackBarHarness);
+    expect(await snackBar.getActionDescription()).toBe('Unsubscribe');
+
+    // For snack-bar's with custom template, the action description
+    // cannot be retrieved. We expect an error to be thrown.
+    fixture.componentInstance.openCustom();
+    snackBar = await loader.getHarness(snackBarHarness);
+    await expectAsyncError(() => snackBar.getActionDescription(), /custom content/);
+  });
+
+  it('should be able to check whether simple snack-bar has action', async () => {
+    fixture.componentInstance.openSimple('With action', 'Unsubscribe');
+    let snackBar = await loader.getHarness(snackBarHarness);
+    expect(await snackBar.hasAction()).toBe(true);
+
+    fixture.componentInstance.openSimple('No action');
+    snackBar = await loader.getHarness(snackBarHarness);
+    expect(await snackBar.hasAction()).toBe(false);
+
+    // For snack-bar's with custom template, the action cannot
+    // be found. We expect an error to be thrown.
+    fixture.componentInstance.openCustom();
+    snackBar = await loader.getHarness(snackBarHarness);
+    await expectAsyncError(() => snackBar.hasAction(), /custom content/);
+  });
+
+  it('should be able to dismiss simple snack-bar with action', async () => {
+    const snackBarRef = fixture.componentInstance.openSimple('With action', 'Unsubscribe');
+    let snackBar = await loader.getHarness(snackBarHarness);
+    let actionCount = 0;
+    snackBarRef.onAction().subscribe(() => actionCount++);
+
+    await snackBar.dismissWithAction();
+    expect(actionCount).toBe(1);
+
+    fixture.componentInstance.openSimple('No action');
+    snackBar = await loader.getHarness(snackBarHarness);
+    await expectAsyncError(() => snackBar.dismissWithAction(), /without action/);
+  });
+}
+
+/**
+ * Expects the asynchronous function to throw an error that matches
+ * the specified expectation.
+ */
+async function expectAsyncError(fn: () => Promise<any>, expectation: RegExp) {
```

Comment:
is it worth moving this into cdk/testing or cdk/private/testing?

Reply:
Yes. moved into `cdk/private/testing`.